cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -74,12 +130,66 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Sch
     throws SchedulerException {
     Preconditions.checkArgument(schedule instanceof StreamSizeSchedule,
                                 "Schedule should be of type StreamSizeSchedule");
+    StreamSizeSchedule streamSizeSchedule = (StreamSizeSchedule) schedule;
+    schedule(program, programType, streamSizeSchedule, true, -1, -1, true);
+  }
+
+  /**
+   * Handle a new {@link StreamSizeSchedule} object in this scheduler.
+   *
+   * @param program Program that needs to be run
+   * @param programType type of program
+   * @param streamSizeSchedule Schedule with which the program runs
+   * @param active {@code true} if this schedule is active, {@code false} otherwise
+   * @param basePollSize size, in bytes, used as the base count for this schedule, or -1 to start counting from
+   *                     the current size of events ingested by the stream, as indicated by metrics. Another way
+   *                     to see it is: size of the stream during which the schedule last executed the program,
+   *                     or -1 if it never happened yet
+   * @param basePollTs timestamp, in milliseconds, which matches the time at which {@code basePollSize} was computed.
+   *                   -1 indicates to start counting from the current timestamp
+   * @param persist {@code true} if this schedule should be persisted in the persistent store containing the
+   *                stream size schedules, {@code false} otherwise
+   */
+  private void schedule(Id.Program program, SchedulableProgramType programType, StreamSizeSchedule streamSizeSchedule,
+                        boolean active, long basePollSize, long basePollTs, boolean persist) throws SchedulerException {
+    // Create a new StreamSubscriber, if one doesn't exist for the stream passed in the schedule
+    Id.Stream streamId = Id.Stream.from(program.getNamespaceId(), streamSizeSchedule.getStreamName());
+    StreamSubscriber streamSubscriber = streamSubscribers.get(streamId);
+    if (streamSubscriber == null) {
+      streamSubscriber = new StreamSubscriber(streamId);
+      StreamSubscriber previous = streamSubscribers.putIfAbsent(streamId, streamSubscriber);
+      if (previous == null) {
+        try {
+          streamSubscriber.start();
+        } catch (NotificationFeedException e) {
+          streamSubscribers.remove(streamId);
+          LOG.error("Notification feed error for streamSizeSchedule {}", streamSizeSchedule);
+          throw new SchedulerException(e);
+        } catch (NotificationFeedNotFoundException e) {
+          streamSubscribers.remove(streamId);
+          LOG.error("Notification feed does not exist for streamSizeSchedule {}", streamSizeSchedule);
+          throw new SchedulerException(e);
+        }
+      } else {
+        streamSubscriber = previous;
+      }
+    }
+
+    // Add the scheduleTask to the StreamSubscriber
+    if (streamSubscriber.createScheduleTask(program, programType, streamSizeSchedule,
```

Comment:
not sure when this would return false. One case I know is if the schedule already exists. Then you should throw an exception, not silently ignore the attempted overwrite.